src/App.jsx

```diff
@@ -14,70 +15,88 @@ export const goods = [
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+        {selectedGood && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => {
+              setGood('');
+            }}
+          />
+        )}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+      <table className="table">
+        <tbody>
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+          {goods.map(el => (
+            <tr
+              className={selectedGood === el
+                ? 'has-background-success-light'
+                : null}
+              data-cy="Good"
+              key={el}
+              id={el.split(' ').join('')}
             >
-              +
-            </button>
-          </td>
+              <td>
+                {selectedGood === el
+                  ? (
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={() => {
+                        setGood('');
+                      }}
+                    >
+                      -
+                    </button>
+                  )
+                  : (
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className="button"
+                      onClick={() => {
+                        highlightGood(el.split(' ').join(''));
+                        setGood(el);
+                      }}
+                    >
+                      +
+                    </button>
+                  )
+                }
+              </td>
+              <td data-cy="GoodTitle" className="is-vcentered">
+                {el}
+              </td>
+            </tr>
+          ))}
+        </tbody>
+      </table>
+    </main>
+  );
+};
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+const highlightGood = (id) => {
+  document.querySelectorAll('tr')
+    .forEach(el => el.classList.remove('has-background-success-light'));
+  document.querySelector(`#${id}`).classList.add('has-background-success-light');
```

Comment:
don't use `document` in React application
if you need to make a dynamic className you can do it in a template